### ISSUE TITLE
App release

### ISSUE THREAD

**Issue by brnwlshubh** [z-huang/music#1]
Please release the app on Fdroid

**Comment by z-huang** [comment]
Thanks for your advice. However, in the current stage, the project isn't stable. There are so much more functions to be completed to make this application really useful for music listening, such as customizable settings, more user-friendly interface, showing lyrics, embed youtube-dl as a second backend, etc. Also, I'm not sure when I can make a stable release. Therefore, it may not be the right time to publish this app to F-droid now. However, I can implement an updater inside the app to automatic check updates and download the apk file, which may be more convenient for users to continue receive updates.

**Comment by IzzySoft** [comment]
Hi @z-huang – I've took the freedom to add your app [to my repo](https://apt.izzysoft.de/fdroid/index/apk/com.zionhuang.music), which often serves as "stepping stone" for inclusion with F-Droid (or to provide betas where F-Droid serves the stable ones). You're welcome to [use a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to there e.g. from your Readme.

I've also opened a PR providing you initial Fastlane structures. If you accept that, I'll toggle the switch in my repo to sync that whenever pulling updates. As described in the PR, this gives you a.o. the option to add a paragraph to the description stating the app is not yet "stable" – and remove that part again once it matured.

That said: my library scanner says it should be fine to apply for inclusion at F-droid.org – no show stoppers found. Thanks a lot for keeping it that clean!

PS: if you wonder about the NonFreeNet mark (which will show up tomorrow, I forgot to apply it in time for today's sync) – that's just because your app depends on YT (you will find the very same mark on the NewPipe app itself), so nothing to worry about. People would wonder if it were not there :wink:

**Comment by z-huang** [comment]
I'm happy to know this way of publishing my app and thanks for your doing so. It's much easier than publishing to the F-Droid repo. Therefore, I've merged your PR. I'll update the Readme file to add information such as "unstable" and an IzzyOnDroid badge.
Thanks for your inclusion to your repo!

**Comment by IzzySoft** [comment]
Gladly – and thanks for accepting my PR! I've just updated the config at my end to pull updates from there whenever a new version is pulled.

Once you reach a state you'd consider mature and want your app in the "official" F-Droid repo: if you need help with that, we'll certainly meet again and I gladly assist. Disclosure: I'm one of the F-Droid maintainers :wink:

> I'll update the Readme file to add information such as "unstable"

This part you'll probably also want to add to the `full_description.txt` in your new Fastlane structures for now (and remove it again later). Please remember the [formatting hints](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Fastlane) for that file so it then looks the way you want it to :smiley:

> and an IzzyOnDroid badge

Great, thanks! Looking forward to that :smiley:

**Comment by z-huang** [comment]
I've update the `README.md` and `full_description.txt` file to add the badge and the "unstable stage" note.
I'm grateful for your inclusion to your repo, for it brings a lot of visitors to this project.😊

> Once you reach a state you'd consider mature and want your app in the "official" F-Droid repo: if you need help with that, we'll certainly meet again and I gladly assist.

So kind of you!😊

**Comment by IzzySoft** [comment]
Gladly – and thanks! I've just manually triggered an update from Fastlane, so with today's sync your changes will become visible.

**Comment by z-huang** [comment]
@IzzySoft Since this app has a new name and is enough for daily use, I'd like to publish it to F-Droid. Is there something I need to do?

**Comment by IzzySoft** [comment]
Just open a [request for packaging](https://gitlab.com/fdroid/rfp/-/issues) and it should be picked up there. If the bot approves, you can also open the corresponding merge request with the YAML metadata at fdroiddata to speed up things (I might join in at that point to help out a bit).

**Comment by z-huang** [comment]
@IzzySoft Why isn't this app showing in IzzyOnDroid?

**Comment by IzzySoft** [comment]
@z-huang because it was added to the F-Droid.org repo on 2022-11-05 (once that happens and given an overlap of approx. 2 weeks), apps are removed from my repo unless explicitly requested to stay.

Looking at the F-Droid.org data, I see your app is using reproducible builds – so cross-updates would be possible and no "irritations" because of signature mismatches. Shall I take your question as request to re-establish your app in my repo? Can be done easily as I just archived the metadata here. I'd then add it back and place the "keep" marker on it.
